### PR TITLE
Anfield feature: Support for Global whitelists, Vlans and Vcenter settings.

### PR DESCRIPTION
**Pull request by naveena-maplelabs**
Added support for following features,

1. Vlans import and export along with intergace groups.
2. Global whitelists settings
3. Vcenter Source settings like throttling policy, policy overrides and minimum free space and subnets.

### CODE REVIEW

**Comment by naveena-maplelabs** [comment]
@pyashish Can you please review.

**Review by pyashish — APPROVED** [review]

File: samples/project-anfield/export_cluster_config.py [hunk 1]
```diff
@@ -87,6 +88,34 @@
...
+    """
+    try:
+        settings = dict()
+        subnets = cohesity_client.clusters.get_external_client_subnets(\
+            ).client_subnets
+        settings["subnets"] = subnets if subnets else []
+        exported_res["Subnet whitelists"] = [
+            subnet.ip for subnet in settings["subnets"]]
+
+        api = "nis-providers"
```
[the remark this answers is not in the document]
Comment (naveena-maplelabs) [reply samples/project-anfield/export_cluster_config.py]: Added Constants

**Comment by naveena-maplelabs** [comment]
@pyashish Added support for vlans.

File: samples/project-anfield/export_cluster_config.py [hunk 2]
```diff
@@ -129,7 +128,9 @@ def get_whitelist_settings(cohesity_client):
     "sql_entity_mapping": library.get_sql_entity_mapping(cohesity_client,
                                                          env_enum.KSQL),
     "ad_entity_mapping": library.get_ad_entity_mapping(cohesity_client,
-                                                         env_enum.KAD)
+                                                         env_enum.KAD),
+    "vlans": library.get_vlans(cohesity_client),
```
Comment (pyashish) [inline samples/project-anfield/export_cluster_config.py:132]: something doesn't add up: 
1. while exporting you export all these by default 
2. While importing you import selectively. 

What if there is sensitive information and customer doesn't want to export it by default ?

File: samples/project-anfield/import_cluster_config.py [hunk 3]
```diff
@@ -625,7 +716,7 @@ def create_protection_sources():
 
         for source in sources:
             environment = source.protection_source.environment
-            if environment not in env_list:
+            if environment not in [env_enum.K_VMWARE]: #env_list:
```
Comment (pyashish) [inline samples/project-anfield/import_cluster_config.py:719]: Make sure this doesn't regress other env type, why did we restrict it to vmware only ?

File: samples/project-anfield/import_cluster_config.py [hunk 4]
```diff
@@ -1306,6 +1397,41 @@ def create_roles():
...
+        except Exception as err:
+            ERROR_LIST.append(
+                "Error while creating interface group %s, err msg: %s" % (
+                    group.name, err))
+    existing_vlans = [
+        vlan.iface_group_name for vlan in library.get_vlans(cohesity_client)]
+    exported_vlans = cluster_dict.get("vlans", [])
+    for vlan in exported_vlans:
+        try:
+            import pdb;pdb.set_trace()
```
Comment (pyashish) [inline samples/project-anfield/import_cluster_config.py:1421]: this will stop the code ? Big red flag !

**Review by pyashish — CHANGES_REQUESTED** [review]

**Pull request merged**